### ISSUE TITLE
TV Nowa

### ISSUE THREAD

**Issue by MajkiIT** [X01X012013/AdBlockProtector#23]
Hi, can you help me with this page?

http://dziendobry.tvn.pl/wideo,2064,n/proste-cwiczenia-na-poprawe-krazenia,3193.html

**Comment by X01X012013** [comment]
Hi, This is patched into 1.2.3
```JavaScript
onEvent("load", function () {
    $("#dl_link").show();
    $("#adblock_msg").remove();
});
```

**Comment by X01X012013** [comment]
Hi, please try this workaround: 
`
@@||*.tvn.pl/*^$generichide
@@*ad.xml*$domain=tvn.pl
@@|http://imasdk.googleapis.com/js/sdkloader/ima3.js|$domain=tvn.pl
@@|http://s1-player5.cdntvn.pl/adscript.js*$domain=tvn.pl
@@*TVN-Adserver*$domain=tvn.pl
`

**Comment by MajkiIT** [comment]
@X01X012013 I have modyfication of your script. http://pastebin.com/dMp1HpP8 It works if you choose quality of video. Can you fix it?

https://cloud.githubusercontent.com/assets/18214146/19934802/41ed2640-a118-11e6-960d-3fca4b73ec5a.PNG

**Comment by X01X012013** [comment]
Hi, thank you for your contribution, this is patched into 1.2.6.

**Comment by xxcriticxx** [comment]
@X01X012013 can you do something to suppress the ads from showing on tvn.pl?

**Comment by X01X012013** [comment]
@MajkiIT I don't think simply locking that variable will work, did I do something wrong?

**Comment by MajkiIT** [comment]
@X01X012013 this script is not my. Someone show me it. Also I'm not a programmer. I show you this script, because you will modify it.

**Comment by xxcriticxx** [comment]
@X01X012013 we got some instructions please read this over see if it make sense

Sadly I don't have enough time now but I looked quickly and the best way to do would be to replace the player file: http://s1-player5.cdntvn.pl/player.js to edit it and remove the addblock checker inside.

After you would have just to make a Userscript to stop the file from being loaded and replace player.js with the fixed player.js that don't check for addblocker.

This would only work with Firefox since it would be done with the checkForBadJavascripts() function. :octocat:

Here's a post that explain how it work to block a script and change it with another: http://stackoverflow.com/questions/10472569/changing-javascript-on-an-html-page-out-of-my-control.

Good Luck and let me know if you are successful, I'm curious to know :smiley_cat:

**Comment by X01X012013** [comment]
@xxcriticxx All we really need to to put `http://s1-player5.cdntvn.pl/player.js` into AdBlocker's blacklist, and load the good one from somewhere. The problem here is who want to modify that 50 **thousand** lines `player.js`.

**Comment by xxcriticxx** [comment]
or you can check for `checkForBadJavascripts()` function whatever that means

**Comment by X01X012013** [comment]
@xxcriticxx 
`checkForBadJavascripts()` function is a custom function and is not built into any browser. It relies on `beforescriptexecute` event, however, none of that is needed to replace `player.js`. What we need is the patched version of `player.js`.

**Comment by X01X012013** [comment]
@xxcriticxx 
Maybe I'm lazy, but I really don't want to go over `50983` lines of code just to block some pre-rolls until `player.js` gets updated... Maybe I'll do it later if I feel like it...

**Comment by xxcriticxx** [comment]
@X01X012013 got it so we have workaround for right now with short 30 sec commercial

**Comment by X01X012013** [comment]
@xxcriticxx 
Yep, the rules in the second post from the top will do that, I will comment out this website from the script so the rules will work.

**Comment by mikhoul** [comment]
@X01X012013  It was my idea to `checkForBadJavascripts() `since you can change **both** INLINE or remote JS script with this function with no need for an external blocker. It's more easy to debug IMO when you have everything in the same script. :money_mouth_face: 

I was browsing this thread and I agree with you that 50K lines of code is a lot 😜.

But I was thinking if we could use a simple player version with less protection against adblocker maybe it will have only the player part of the code, so it would be lot less code...  :octocat: 

I said to myself let's try to find an old version of the player.js... so Google  is my friend I found something interesting, 2 links: 

http://playerjs.io/
https://github.com/embedly/player.js

it would be worth just to try to replacing the big bloated player.js with this one, it could be the original without protection against addblockers.  😊

I'm lot less pro-efficient than you with JS but let me know if it's work, I would install the whole files on host somewhere  like https://infinityfree.net, I use them to host my JS scripts when I want them to be remote. 

Anyway let me know how it turn !  ✌️

**Comment by X01X012013** [comment]
@mikhoul 
`checkForBadJavascripts()` is great but it only work on FireFox, this project is designed for Opera and Chrome so `checkForBadJavascripts()` and `beforescriptexecute` cannot be used. 
I didn't know there is a project called `player.js` so thought `*.tvn.pl` made `player.js` themselves. If we start from here it is much simpler, but it is still a LOT of work since `*.tvn.pl` seems to have customized the  `player.js` that they are using. 
I will first find a way to patch `Function.prototype.toString()` that might be required to fix `timesofindia.indiatimes.com` when it breaks again, then I will have a look at this issue.

**Comment by xxcriticxx** [comment]
sound like fun

**Comment by X01X012013** [comment]
@mikhoul 
Is that hosting service stable? Every single free dynamic website hosting service I tried is rather useless. I'm currently hosting everything on GitHub.

**Comment by mikhoul** [comment]
@X01X012013   

>  Is that hosting service stable?   

It seem really stable, before using this service I tried over the years 5-6 other "free" hosting service but all had to much limitations for me like: To keep your account you must log-in once a month or your account was deleted, or no support for PHP etc... :computer: 

The only limitation I've seen from this one is the upload speed to prevent the sharing of Software/movies etc..  but it's enough for me. It support PHP, FTP and with FTP I use it with WinSCP so I can edit and save my file directly in Notepad++ and everything is updated in real time. :smile_cat: 

I use it since a few months and everything is fine till today, the support is VERY good for a free service you got answer within few hours and it's quality support. :octocat: 

Have a Great Day :sun_behind_small_cloud:

**Comment by X01X012013** [comment]
@mikhoul 
The `player.js` from your links only have `1187` lines of code... Did I miss something?

**Comment by mikhoul** [comment]
@X01X012013   No I was surprised myself too but from what I've read it seem that "player.js" is a kind of standard between websites, so the code added seem just to be for advertisement/adblocker.

Even if it was not the same exactly I was telling to myself maybe it would be worth to replace the player completely and connect the new "clean" player directly to the stream of the video. 

This way we could bypass the Evil... 😈

**Comment by xxcriticxx** [comment]
they added 49k lines of code to it wow

**Comment by X01X012013** [comment]
@mikhoul 
Their script calls `video.player5` to initialize the video, which is not defined in `player.js`, could it be from some other libraries? Like `video.js`?

**Comment by xxcriticxx** [comment]
different version?

**Comment by X01X012013** [comment]
@xxcriticxx 
No, I don't think so, it's most likely to be a fully customized `player.js`.

**Comment by xxcriticxx** [comment]
@X01X012013 tvn is big company am sure they have some money

**Comment by X01X012013** [comment]
@xxcriticxx 
If they are big, then it is likely someone already developed a patch...

**Comment by mikhoul** [comment]
@X01X012013   Ok  I studied/digged a little bit more the "logic" of the code as a whole and I found the the player is spread around 2 main files at least:   
![GitHub Logo](http://i.imgur.com/THQzUlJ.png)
 ![](url)


The player had multiple options inside the code do all kind of things other than playing video... so since  I'm not really pro-efficient in JS  it's hard for me understand but even if I was good at JS it is REALLY big and rely a lot on external frameworks.  

**The good news:**  I found a relatively easy way to bypass the player and skip the all advertisements.  😺

Each HTML page content has the direct link to the video with the token to be able to see the video.

It's easy to find, all their videos are in .MP4 so you just search for `data-src=` and next to it you have the URL with the token like: http://r.dcs.redcdn.pl/http/o2/tvn/web-content/m/v/7c9d0b1f96aebd7b5eca8c3edaa19ebb/3340c46c-cfb8-11e0-9d8d-0025b511229e-360p.mp4?salt=FD976E7AB898CE93B2DFBBCBE8E51BE7&token=739769450ACE67F2B4D9C01F7486A2E18FAF6D5A9BF08702AEF5C384083248DC34F5DBAEE2FF96EBA060C7DF50274109D7A1D6D8EDD8583CFDDA90F43BBE69314248ED196908B3C4C7048A39C1EEFCB0D3C5E8456EF4192BA3FC09964A1E7DD5299106B675CC3EBA5747E50F06A49C8D608369468400CB79FBD8FDFF09A7D3C3

![GitHub Logo](http://i.imgur.com/inFAlJD.png)
 ![](url)


Videos are always on hosted on: http://r.dcs.redcdn.pl

The logic is simple: 

1. Stop the 2 big scripts for the video player from being loaded.
2. Scrap the web page for the URL of the video via `data-src=` 
3. Inject the new inline HTML/CSS/JS  with the URL for the new Video Player in the web page

That's it !  😄

✨Here's a **JsBin prototype** I've setup with an easy and light JS Video player with all the player code:
                                     💫 http://jsbin.com/vigare/edit?html,output 💫

Just copy and paste all the scripts are hosted on CDN.

This way nobody should have to care about future change in the code fromvideo player.  😜
You could even use this code after for any problematic video player that display ads.

Let me know how it turn :octocat:

**Comment by xxcriticxx** [comment]
@mikhoul nice job! you went bit crazy with the emojis :laughing:

**Comment by xxcriticxx** [comment]
@mikhoul can you check other tvn websites?

`http://www.tvn.pl/`
`http://www.tvn7.pl/`
`http://www.tvnstyle.pl/`
`http://www.tvnturbo.pl/`
`http://tvnfabula.pl/`
`http://u.itvn.pl/`
`http://itvnextra.pl/`
`http://www.tvn24.pl/`
`http://tvnwarszawa.tvn24.pl/`
`http://tvn24bis.pl/`
`http://tvnmeteo.tvn24.pl/`
`http://tvnmeteoactive.tvn24.pl/`
`http://ttv.pl/`
`http://player.pl/`
`http://kulinarni.tvn.pl/`
`http://kontakt24.tvn24.pl/`
`http://x-news.pl/login`

**Comment by X01X012013** [comment]
@mikhoul 
WOW thanks a lot, I'll try this out. I knew I wasn't searching at the right place for the key...

**Comment by X01X012013** [comment]
GOOD NEWS! This website is fixed in 1.2.15!

**Comment by X01X012013** [comment]
@xxcriticxx I will now have a look at your websites, I will update this post as we go. 
```
http://www.tvn.pl/
http://www.tvn7.pl/
http://www.tvnstyle.pl/
http://www.tvnturbo.pl/
http://tvnfabula.pl/
http://u.itvn.pl/
http://itvnextra.pl/
http://www.tvn24.pl/
http://tvnwarszawa.tvn24.pl/
http://tvn24bis.pl/
http://tvnmeteo.tvn24.pl/
http://tvnmeteoactive.tvn24.pl/
http://ttv.pl/
http://player.pl/
http://kulinarni.tvn.pl/
http://kontakt24.tvn24.pl/
http://x-news.pl/login
```

**Comment by mikhoul** [comment]
> @mikhoul can you check other tvn websites?   @X01X012013  @xxcriticxx 

Yes it will work on all their website since they use the same scheme for the structure of the HTML pages.

Some vide don't have TOKENS just a plain URL so IMO it's more simple to play all videos in "our" player than parsing the videos to see if they have or not Tokens. :octocat: 


:smile_cat:

**Comment by X01X012013** [comment]
@mikhoul Sorry to disappoint you but that's not really how it works... The structure is slightly different. If someone can give me links to actual videos inside instead of home page, that would help a lot.

**Comment by xxcriticxx** [comment]
@X01X012013 

http://www.tvnturbo.pl/aktualnosci,1850,n/najwiekszy-statek-pasazerski-i-najdluzszy-serial,216184.html
http://www.tvnturbo.pl/aktualnosci,1850,n/jak-rozwinac-skrzydla-w-bialymstoku,211664.html

**Comment by xxcriticxx** [comment]
@X01X012013 

http://www.tvnstyle.pl/aktualnosci,1865,n/rusza-casting-do-programu-sekrety-lekarzy,181940.html
http://www.tvnstyle.pl/aktualnosci,1865,n/paulina-krupinska-chciala-zjesc-w-peru-swinke-morska,215964.html

**Comment by X01X012013** [comment]
@xxcriticxx 
Hi, both are fixed in 1.2.18. I'm still working on home pages.

**Comment by mikhoul** [comment]
@X01X012013   I tried your userscript but it did not work and pretty sure I miss some part of the "software" since I don't see the code injected in the page for the new player ?  😳


I looked at 3 other pages and they all have the same structure/format !

I will try again to explain more clearly how I see the pseudo-code for the workaround, it was late last night maybe I had some fog in my mind  😏

- The first thing to do is to search with a REGEX for an URL with or witout token that BEGIN with `data-src= "http://`   AND that include `redcdn.pl/dcs/`  AND that have after `.mp4`  AND end with    `"`  
 
- If it is found we copy the URL with the token if available.
 
- We remove the div `.videoPlayer`  to remove the player
 
- We inject the code for the new MediaElement video player with the URL in the page.
 
- Optionally to save loading time we could block all files for the VideoPlayer since they are big and no longer used.

I found those 2 scripts that seem to have already some useful function inside that could be used:
https://greasyfork.org/en/scripts/8684-change-video-player-for-vkmag/code
https://greasyfork.org/en/scripts/14476-yahvt/code

Here's one of the video without token (not protected)
http://n-4-1.dcs.redcdn.pl/dcs/o2/tvn/web-content/m/p1/v/89f03f7d02720160f1b04cf5b27f5ccb/30e920f0-f54d-408b-aa68-77024404967f-720p.mp4

Another one with a token (protected): http://n-0-19.dcs.redcdn.pl/dcs/o2/tvn/web-content/m/p36/v/10c272d06794d3e5785d5e7c5356e9ff/2481247f-ec55-44d1-badd-561c1b079702-480p.mp4?salt=E0A373C22D9C39475F03B53E7D003C83&token=4D920D80DA86424EF7E16EBAE292147BA1B3FE47C1068F92BB1B6621FF34E7AE71355900D4999B8B22B8DD59BB5E74523F9B59A0D3A560A3E92E9D560067616DD65CB4D19ABC2AE435F2F60D73F52854DB46A539005332EC2F0F0253FB6EB2FD248478CCDF28D104FE8F569A8BB0EDE30087E13BDFDAE5A045B949DF639499D8926B8339BD2BF57ABF863A2862324602

Let me know what you think about it, if it si more clear... 🐸

**Comment by X01X012013** [comment]
@mikhoul 
I am using `Array.prototype.includes` to filter domain, which browser are you using? 
I used `$(".videoPlayer5").data("src")` to grab the source. 
I inserted an `iframe` and just let the browser handle the rest, it works fine on Opera. 
Home pages are still work in progress. 
![screenshot](http://i.imgur.com/NDLYL56.png)


Thank you for your help 😄

**Comment by X01X012013** [comment]
@mikhoul Any clue on how to get started on fixing their home pages?

**Comment by xxcriticxx** [comment]
user script?

**Comment by xxcriticxx** [comment]
@X01X012013 why did you change title of this post?

**Comment by mikhoul** [comment]
@X01X012013  I use Firefox mainly,  I was not knowing what `Array.prototype.includes`  was doing  lol 🙀
After reading quickly at Javascript MDN I understand a little bit more.  I will have to follow a basic course in JS to fully understand the basic of JS and I will lose less time than trying to figure all the things by myself.    

Firefox can handle fine HTML5 video like Chrome but both don't have fullscreen option and don't handle well the different screensize for the video depending on their original encoding.

Since they are TV show/news to said to myself that the user like me like to see their TV show in full resolution with a better Video player. That's why I tested the MediaElement video player on both Firefox and Chrome to be sure it handle well the different format of screen and also Fullscreen.

I understand it's a little bit more complicated to integrate a new video player than using the builtin but once the code is done it's for all videos for the next years... 😏  

If you wrote the code with the way I see it the only thing to change in the future for any other websites would be the REGEX for the URL and the name of the DIV to replace the video player. So on the long run it could make the things more easy to maintain and reuse the same code for other websites.

😊

**Comment by mikhoul** [comment]
> @mikhoul Any clue on how to get started on fixing their home pages? 

@X01X012013 Sorry I did not see this post.

Could you give me the URL of the homepage I'm not sure of which homepage we are talking about ?

I will take a look but I'm pretty confident that if we use a REGEX to find the URL with the conditions mentioned in my previous post it will work on any page of TVN. Also the div `.videoPlayer` to remove the player is always the same on everypage with a player so that's why I would use REGEX and only delete the same DIV to replace it with "our player".

It's alittle bit more coding but once it is done it will work on every pages of all those TVN domains.  📺


:octocat:

**Comment by X01X012013** [comment]
@xxcriticxx Since this post is about all TVN websites, I thought this would be a clearer title. 

@mikhoul You are right, making it a library function will make patching other websites easier, but I still don't see the point of using RegExp to find URL when you can just use `$(...).data(...)` which is much more efficient and easier. 
For home page, I tried to do the same but the structure is quite different and we need to figure out a new place to inject the video. 
http://www.tvn.pl/
![screenshot](http://i.imgur.com/Fs4LsCk.png)

**Comment by X01X012013** [comment]
@mikhoul 
Maybe we can borrow this? 
https://github.com/reek/anti-adblock-killer/blob/master/anti-adblock-killer.user.js#L1040

**Comment by mikhoul** [comment]
@X01X012013  I will look at the code deeper tonight  and come back later, in meantime double-check with your scripts it prevent to use this fiddle: http://jsfiddle.net/N8Zs5/18/#run  😊

**Comment by X01X012013** [comment]
@mikhoul 
What do you mean by "prevent to use this fiddle"? That fiddle behave the same with or without AdBlock Protector enabled.

**Comment by mikhoul** [comment]
@X01X012013 

> What do you mean by "prevent to use this fiddle"? That fiddle behave the same with or without AdBlock Protector enabled. 

 Try under Firefox: http://screencast-o-matic.com/watch/cDX1nijL6r  😺

> I still don't see the point of using RegExp to find URL when you can just use $(...).data(...) which is much more efficient and easier. 

Use the method that you think will be the more easy to maintain in the future and the less prone to break later. It's hard for me to tell since I don't understand fully how work `Array.prototype.includes` .

For the main/home page: http://www.tvn.pl/ they are only small video previews/advertisements/teasers for the TV show, not real episodes. So I would just remove the overlay with simple filter like www.tvn.pl###_videoDriverContent4 (1-2-3-4-5-6-7-8 depending on how much preview they have.). 

If they want to see the show they click on the link beside and it bring them to the portal of the show where the full episodes of the show are. Also on all those page it's the same layout.  

IMO it would be to much work to try to fix the video players on the main page since they are only short advertisements/teasers to bring users to listen the full episodes with the links.

> Maybe we can borrow this?
> https://github.com/reek/anti-adblock-killer/blob/master/anti-adblock-killer.user.js#L1040

I took a look a the code, most of the code seem to be for flash video which is less and less used today, it seem to have many many fallback in case one video player won't work.

IMO we don't need such fallback the best is to keep it simple, last night I tried numerous video players to see how they work, what advantage they have, the compatibility, from my test  MediaElement video player has given the best results for simplicity and compatibility for what we need. Other video players have lot more features (like overlay inside, hidden link, ,Skins, tracking of the users etc etc) but features we will not use here. Also video element is actively maintained.

If you are able to make your own small JS library Video player it will be more easy in the future than rely on REEK script that will become obsolete the more the time pass. Also you if something change or the need change in the future it will be more easy if it is your own code to understand.

But you can look for inspiration if you see good ideas, concept use it no need to reinvent the wheel each time. 😏 

Soon I will try to debug Slacker 🎵 completely with a more permanent option and will let you know of my solution and if you want to integrate it in your script. 

Have a good night 🌛

**Comment by X01X012013** [comment]
@mikhoul 
Sorry, currently, I do not offer support for FireFox, the link works fine on Chrome and Opera. 
Every website will have different way of storing the "token", it will be case by case, but for TVN, use jQuery's `data(...)` is the best way to go. 
I use `Array.prototype.includes` to see if the domain is in a list, it is supported by both Chrome and Opera. It can cause issues on Edge, but no one uses Edge anyway. 


There is another issue that we are facing, that is Content Security Policy, it prevents external and in-line scripts from being loaded on the fly, I believe it does not affect `iframe`, but then we will have issues determining how high the `iframe` needs to be. 


To make delivering rules easier, I created AdBlock Protector List 😄 


Again, thank you for your time and effort 👍

**Comment by MajkiIT** [comment]
Latest version block films on youtube.

**Comment by X01X012013** [comment]
@MajkiIT 
Sorry, I cannot reproduce this issue, do you have a specific link to a broken page? 
![screenshot](http://i.imgur.com/5ozFbpu.png)

**Comment by X01X012013** [comment]
Good news, Reek is coming back, since our fix is acceptable, we will stop here and wait for Reek for the permanent / semi-permanent fix.

**Comment by xxcriticxx** [comment]
@X01X012013 how did you get this news?

**Comment by mikhoul** [comment]
@X01X012013 

> Good news, Reek is coming back, 

Are you sure ?   🙀

**Comment by X01X012013** [comment]
@xxcriticxx @mikhoul 
https://github.com/reek/anti-adblock-killer/issues/1783#issuecomment-258712811

**Comment by xxcriticxx** [comment]
lets hope that is him and no body crack his nick password

**Comment by mikhoul** [comment]
I just received the update for the latest commits ! :smile:

**Comment by xxcriticxx** [comment]
same here!

**Comment by MajkiIT** [comment]
@X01X012013 FF on ubuntu
![zrzut ekranu z 2016-11-07 20-38-00](https://cloud.githubusercontent.com/assets/11302675/20072855/60ddd4ee-a52a-11e6-8947-f20b2b18636c.png)

**Comment by xxcriticxx** [comment]
i test same flick on ubuntu 16.04 same as @MajkiIT got black screen only

**Comment by X01X012013** [comment]
@MajkiIT @xxcriticxx 
Can you guys send me your console output? 
We don't officially support FireFox, but I can have a quick look.

**Comment by xxcriticxx** [comment]
![capture](https://cloud.githubusercontent.com/assets/15007183/20074162/1cf478d2-a4fd-11e6-8e61-50176490efe8.PNG)

**Comment by X01X012013** [comment]
Fine. I will install FireFox and have a look.

**Comment by X01X012013** [comment]
OK, I installed FireFox, and everything works just fine on it. Maybe update your FireFox? 
![screenshot](http://i.imgur.com/Y7InI6x.png)

**Comment by xxcriticxx** [comment]
@X01X012013 try with greasemonkey and ublock origin

**Comment by mikhoul** [comment]
@X01X012013  I'm pretty sure I know where the problem is let me test it and in a few minutes I will be back with the offending line in the code.

**Comment by X01X012013** [comment]
@xxcriticxx 
All my code are designed for Tampermonkey and Adblock Plus, you should consider switching setup.

**Comment by mikhoul** [comment]
>  you should consider switching setup. 

Not for me, UBO is FAR MORE powerful than ABP, I hate the UBO GUI but it is lot more powerful and same thing for Greasemonkey who is the one that set the standard for the Userscripts. :hear_no_evil:

**Comment by X01X012013** [comment]
@mikhoul 
I will consider your setup, but for now, I won't test on a different setup. 
Greasemonkey only defined a few functions so Userscripts can save settings, the syntax standard is set by ECMA.

**Comment by mikhoul** [comment]
@X01X012013   Greasemonkey can do lot more than TM, I'm not "religious fanatic" about software I just use the more powerful and the one that let me customize it the way I like.

GM even have some experimental version where you can tweak CSP and bypass it and they are considering MAYBE to include it  but it's already there (Open Source) so we can use it. But it's just one thing it have lot more... 

I used ABP till this summer I was not able to do a complex filter without making a Userscript  but with UBO and the advanced conditional function for the filters like `has()` I can do it with a simple filter instead of a script.

I **hate** UBO GUI/Interface it is NOT intuitive at all ABP has a better GUI but at one point you must evolve... to many things miss in ABO versus UBO.  


I will be back with the offending lines in the code for Firefox, I found them but I want to do more testing to understand the code which is long for me... 😜   One of those day I will be better in JS if I can study the basic lol  🙀

**Comment by X01X012013** [comment]
@mikhoul 
Here are some resources that you might like: 
http://www.w3schools.com/js/default.asp
https://www.khanacademy.org/computing/computer-programming
https://developer.mozilla.org/zh-CN/docs/Web/JavaScript

**Comment by xxcriticxx** [comment]
@X01X012013 i have same problem with firefox 49.0.2 + greasemonkey + ubo + win7 youtube has black screen and wont play any vid

second question is @X01X012013 do you ever sleep?

**Comment by X01X012013** [comment]
@xxcriticxx 
Sorry I don't know what's going wrong, everything works fine for me. 
I do sleep.

**Comment by xxcriticxx** [comment]
lets see what @mikhoul provides

**Comment by mikhoul** [comment]
OK ! :smiley:   @X01X012013 @xxcriticxx 

I know the problem lie with something about the `Array.prototype.indexOf` between the version 1.2.20 and  1.3.0 were it was added. Sadly at this time I have not enough knowledge to why it happen under Gresemonkey and not with Tampermonkey.

But I've sent a message to Ariatus with the issue and asked him to explain me "why it happen ?"

So I should receive an answer relatively soon from him :smile_cat:  After that I hope to have a better understanding of why it happen.

Here's the changes that have broken the compatibility: https://github.com/X01X012013/AdBlockProtector/commit/33001a1150ff2b696ab29283c7839c7c5312c633

Regards :octocat:

**Comment by xxcriticxx** [comment]
@mikhoul this pass my skills i will watch from the corner

**Comment by mikhoul** [comment]
@X01X012013 

> "I don't see anything wrong with my code, how else do you compare function pointers? "

I don't know now but I should and hope to know know it very soon...  :wink: 

As soon as I will have some news from Arantius I will let you know.  :octocat:

**Comment by xxcriticxx** [comment]
@mikhoul any news?

**Comment by mikhoul** [comment]
@xxcriticxx  Not at this time, As soon as I receive an answer I will give an update here i the thread  😊 .

**Comment by xxcriticxx** [comment]
thx

**Comment by szymon1118** [comment]
@mikhoul @X01X012013 I found a solution for "player.pl" :D. TVN has its own API, what I learned from this thread: http://miniskrypt.hubaiitv.pl/thread-54-page-25.html and in player.pl page source (example: http://player.pl/programy-online/kuchenne-rewolucje-odcinki,114/odcinek-9,wroclaw-bar-na-szybkiej,S14E09,65000.html) there is `data-video-id=something` and when we add it to the "id" here: `http://api.tvnplayer.pl/api/?platform=ConnectedTV&terminal=Panasonic&format=json&authKey=064fda5ab26dc1dd936f5c6e84b7d3c2&v=3.1&m=getItem&id=something` we can extract video url from JSON and add to the iframe src.

**Comment by X01X012013** [comment]
@szymon1118 
Is that not what I did?

**Comment by xxcriticxx** [comment]
@szymon1118 is tvn the same way?

**Comment by szymon1118** [comment]
@X01X012013 Not really because on *.tvn.pl sites you can easly extract video url directly from site source (as you already did) but "player.pl" use silverlight with DRM protection and you can't extract video url from website source so I think that TVN API is the solution for now.

@xxcriticxx Solution for tvn sites is already done so it isn't necessary to use this method.

**Comment by xxcriticxx** [comment]
@szymon1118 but temp solution plays 30 sec ads right now 

@mikhoul found solution that was pulling out video ID

**Comment by X01X012013** [comment]
@szymon1118 
Ah, I was confused by all these domain names, sorry. 
`player.pl` has Geo lock, so I am not able to test anything. Do you have a link to a video that does not have Geo lock?

**Comment by xxcriticxx** [comment]
@szymon1118 look in the first box

`http://dziendobry.tvn.pl/wideo,2064,n/proste-cwiczenia-na-poprawe-krazenia,3193.html`

**Comment by xxcriticxx** [comment]
brb going on a beer run

**Comment by mikhoul** [comment]
> @szymon1118 look in the first box
> 
> http://dziendobry.tvn.pl/wideo,2064,n/proste-cwiczenia-na-poprawe-krazenia,3193.html
> 

@xxcriticxx 

Again in this page you just take the source anf look for an MP$ file, it will be the video file but since it is "protected" you need to also take the token to be able to see it.

http://r.dcs.redcdn.pl/http/o2/tvn/web-content/m/v/7c9d0b1f96aebd7b5eca8c3edaa19ebb/3340c46c-cfb8-11e0-9d8d-0025b511229e-360p.mp4?salt=64F8763FBFC904B244397E7643B996BF&token=B0AA6A1DD86EFA8680B28FEA4AF8C47CE71E5A085A294A14B14A60492EF4C56F3956FE9910145E41361F36942939FC40900E9009ADCFCA75B70F9197E7A7B50E3228C85C7C2DDCE8F04F89E8176BF2F9883C8F46E9D84E88AD5B01033FB5D3C939C99E02ED641C3EC8A1217BC6205B85214CEB2416357D41D8BE3EE9AD6751F9

One of those days if I 'm less busy maybe I will do a quick userscript to replace the video player with another that you can resize and also use full-screen.

BTW: I did not have any answer for the other question... 

Have a good night !   :smile_cat:

**Comment by xxcriticxx** [comment]
@mikhoul maybe reek will come up with something

**Comment by X01X012013** [comment]
@xxcriticxx Eh, that is not from `player.pl` domain.

**Comment by xxcriticxx** [comment]
@X01X012013 say what?

**Comment by szymon1118** [comment]
@X01X012013 Sorry but It's hard to find any link that isn't geo locked. I modified AAK script but it works only for player.pl as i noticed `http://pastebin.com/tNZbRBiq` Maybe you can improve this somehow?

**Comment by X01X012013** [comment]
@xxcriticxx Your link is not from `player.pl` domain. It is from `tvn.pl`. 
@szymon1118 Which lines are you talking about?

**Comment by xxcriticxx** [comment]
@X01X012013 i know 

@szymon1118 you need link for tvn or player.pl?

**Comment by szymon1118** [comment]
@X01X012013 From 4023 to 4094.

@xxcriticxx No thanks, but you can test more player.pl videos to make sure it's working properly on every video or on almost every.

**Comment by xxcriticxx** [comment]
@szymon1118 i am in usa i be geo locked i wont be able to test much

**Comment by X01X012013** [comment]
@szymon1118 
That code rely heavily on AAK kernel library, because the lack of documentation, it is really hard to understand. Maybe I will have a closer look later, but it would be better to let Reek handle this.

**Comment by szymon1118** [comment]
@X01X012013 @xxcriticxx 
Ok, but now Geo lock should be bypassed. Can you test: `http://pastebin.com/PFecTWk1`. If I failed I'll wait for reek.

**Comment by X01X012013** [comment]
@szymon1118 
Sorry, we will not have Geo lock bypassing implemented into AdBlock Protector.

**Comment by xxcriticxx** [comment]
@szymon1118 what do we test this against?

**Comment by X01X012013** [comment]
This issue is months old, I will close it now.

**Comment by jspenguin2017** [comment]
A real solution is found. Player no longer needs to be replaced.
